internal/system/agent/services.go

```diff
@@ -292,100 +93,33 @@ func getMetrics(services []string) (MetricsRespMap, error) {
-				m.Metrics[service] = msg
-				LoggingClient.Error(msg)
-			} else {
-				m.Metrics[service] = ProcessResponse(responseJSON)
-			}
-			break
-
-		case internal.CoreCommandServiceKey:
-			responseJSON, err := gccc.FetchMetrics()
-			if err != nil {
-				msg := fmt.Sprintf("%s get metrics error: %s", service, err.Error())
-				m.Metrics[service] = msg
-				LoggingClient.Error(msg)
-			} else {
-				m.Metrics[service] = ProcessResponse(responseJSON)
-			}
-			break
-
-		case internal.CoreDataServiceKey:
-			responseJSON, err := gccd.FetchMetrics()
-			if err != nil {
-				msg := fmt.Sprintf("%s get metrics error: %s", service, err.Error())
-				m.Metrics[service] = msg
-				LoggingClient.Error(msg)
-			} else {
-				m.Metrics[service] = ProcessResponse(responseJSON)
-			}
-			break
-
-		case internal.CoreMetaDataServiceKey:
-			responseJSON, err := gccm.FetchMetrics()
-			if err != nil {
-				msg := fmt.Sprintf("%s get metrics error: %s", service, err.Error())
-				m.Metrics[service] = msg
-				LoggingClient.Error(msg)
-			} else {
-				m.Metrics[service] = ProcessResponse(responseJSON)
-			}
-			break
-
-		case internal.ExportClientServiceKey:
-			responseJSON, err := gcec.FetchMetrics()
-			if err != nil {
-				msg := fmt.Sprintf("%s get metrics error: %s", service, err.Error())
-				m.Metrics[service] = msg
-				LoggingClient.Error(msg)
-			} else {
-				m.Metrics[service] = ProcessResponse(responseJSON)
-			}
-			break
-
-		case internal.ExportDistroServiceKey:
-			responseJSON, err := gced.FetchMetrics()
-			if err != nil {
-				msg := fmt.Sprintf("%s get metrics error: %s", service, err.Error())
-				m.Metrics[service] = msg
-				LoggingClient.Error(msg)
-			} else {
-				m.Metrics[service] = ProcessResponse(responseJSON)
-			}
-			break
-
-		case internal.SupportLoggingServiceKey:
-			responseJSON, err := gcsl.FetchMetrics()
-			if err != nil {
-				msg := fmt.Sprintf("%s get metrics error: %s", service, err.Error())
-				m.Metrics[service] = msg
-				LoggingClient.Error(msg)
-			} else {
-				m.Metrics[service] = ProcessResponse(responseJSON)
-			}
-			break
-
-		case internal.SupportSchedulerServiceKey:
-			responseJSON, err := gcss.FetchMetrics()
-			if err != nil {
-				msg := fmt.Sprintf("%s get metrics error: %s", service, err.Error())
-				m.Metrics[service] = msg
-				LoggingClient.Error(msg)
-			} else {
-				m.Metrics[service] = ProcessResponse(responseJSON)
-			}
-			break
-
-		default:
+		if !isKnownServiceKey(service) {
 			LoggingClient.Warn(fmt.Sprintf(">> Unknown service: %v", service))
-			break
+		}
+
+		responseJSON, err := clients[service].FetchMetrics()
+		if err != nil {
+			msg := fmt.Sprintf("%s get metrics error: %s", service, err.Error())
+			m.Metrics[service] = msg
+			LoggingClient.Error(msg)
+		} else {
+			m.Metrics[service] = ProcessResponse(responseJSON)
 		}
 	}
 	return m, nil
 }
+
+func isKnownServiceKey(serviceKey string) bool {
+	switch serviceKey {
+	case internal.SupportNotificationsServiceKey:
```

Comment:
- At first glance, I was looking for the `fallthrough` keyword, which is the directive to fall through to a subsequent case... Maybe it's not required, let's confirm.

Reply:
Yes, this version of the code doesn't work because of what you described. Please see the latest changes.

Reply:
OK, gotcha. I will pull the latest changes.